### DISCUSSION TITLE
Access to JS console

### DISCUSSION THREAD

**Discussion by andrus** [microsoft/vscode-jupyter#16348] [Questions and Answers] [answered]
Hi and happy New Year! 🙂 

Hoping VSCode Jupyter developers and the wider community might have some ideas about the following issue... 

## TL;DR
How do I debug JavaScript produced by the notebook and rendered by VSCode? Is there access to JS console within VSCode Jupyter?

## Details 
The issue is described here - https://github.com/dflib/dflib/issues/410. So this is VSCode with Jupyter connecting to a standalone local Jupyter server with [DFLib JJava Kernel ](https://github.com/dflib/jjava). The notebook is in Java, but this fact is probably not relevant. It uses DFLib Java library, that generates chunks of HTML and JavaScript that require a browser environment to render as a chart. The same notebook works outside VSCode in the browser, but intermittently fails to display a visual chart in VSCode.

I suspect there are JS issues in the generated output related to the VSCode Jupyter environment specifics, but how do I debug this? Is there for instance a way to access the JS console of the notebook? 

## VSCode Environment

Version: 1.96.2
Commit: fabdb6a30b49f79a7aba0f2ad9df9b399473380f
Date: 2024-12-19T10:22:47.216Z
Electron: 32.2.6
ElectronBuildId: 10629634
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Darwin arm64 24.1.0

Jupyter Extension 2024.11.0

**Comment by andrus** [comment, marked as answer]
Ah, found my answer. "Help > Toggle Developer Tools".
